torch/_subclasses/fake_tensor.py

```diff
@@ -2045,6 +2064,67 @@ def go(t: object, real_t: Tensor) -> None:
                 else:
                     tree_map_(go, fake_out, real_out)
 
+                # check aliasing info
+                try:
+                    _check_alias_info(
+                        "Real tensor propagation found an aliasing",
+                        (flat_real_out := pytree.tree_leaves(real_out)),
+                        (real_args, real_kwargs),
+                        (flat_fake_out := pytree.tree_leaves(fake_out)),
+                        (args, kwargs),
+                    )
+                except AssertionError as exc:
+                    raise AssertionError(
+                        "Real tensor propagation found an aliasing mismatch between fake output "
+                        f"and real output for func: {func}"
+                    ) from exc
+
+                for i, (_real_out, _fake_out) in enumerate(
+                    zip(flat_real_out, flat_fake_out)
+                ):
```

Comment:
Why can't we refactor all of https://github.com/pytorch/pytorch/blob/929797dedbf23376123ce95230c01a7e3b71e130/torch/_subclasses/fake_utils.py#L133-L196 into a single function `check_fake_real(fakes, reals)` and call it here? 

If the problem is that that function doesn't have everything we want to check, then we should improve it to check everything that we want to check

Reply:
For example, fake_utils.py has a storage offset check that I don't see here

Reply:
Ah just added. The checks in fake_utils.py were exposing some larger mismatches (bugs?), e.g. strides & requires_grad, so I optionally turned them off for now.

I also added an output value check for non-tensor values - not sure if we should have that in CrossRefFakeMode too?

Lastly it seems like the sizes check in fake_utils.py differs a bit (accomodates for broadcasting, isn't as strict on equality), so I left that alone.